### PR TITLE
Handle op outputs in default asset IO manager

### PR DESCRIPTION
**Pull request by clairelin135**
Addresses https://github.com/dagster-io/dagster/issues/7713

This PR adjusts the default `fs_asset_io_manager` to handle op outputs by storing op outputs in a file directory `/run_id/step_key` to ensure unique directories for each step.

### CODE REVIEW

File: python_modules/dagster/dagster/core/storage/fs_asset_io_manager.py
Comment (OwenKephart) [inline python_modules/dagster/dagster/core/storage/fs_asset_io_manager.py:87]: nit: can this be context.get_run_scoped_asset_identifier?

**Review by OwenKephart — APPROVED** [review]
one comment, otherwise 🚢

File: python_modules/dagster/dagster/core/storage/fs_asset_io_manager.py
Comment (clairelin135) [reply python_modules/dagster/dagster/core/storage/fs_asset_io_manager.py]: Thanks! didn't realize that function existed

**Comment by sryza** [comment]
If I understand correctly, this makes the behavior of fs_asset_io_manager identical to the behavior of fs_io_manager, when the output is not an asset. Thoughts on just merging them together? I.e. getting rid of fs_asset_io_manager and adding logic inside fs_io_manager?  A small advantage is that we'd now have a name that's not as long.

Also, we should merge this after this external contribution lands so that we don't force them to rebase: https://github.com/dagster-io/dagster/pull/8007.

**Review by sryza — CHANGES_REQUESTED** [review]
.

File: python_modules/dagster/dagster/core/asset_defs/assets_job.py
Comment (clairelin135) [reply python_modules/dagster/dagster/core/asset_defs/assets_job.py]: removed ✅

**Comment by clairelin135** [comment]
@sryza I've adjusted this PR to merge functionality of `fs_asset_io_manager` with `fs_io_manager` (and deleted references to `fs_asset_io_manager`). I think this is ready for you to take another look!

**Review by sryza — APPROVED** [review]
Nice! Love to see PRs that remove more lines than they add. We'll need to make sure to message this as a breaking change in our CHANGES.md. Also, as followups, we'll need this for the gcs, s3, and azure IO managers as well.

**Pull request merged**
